opencog/guile/SchemePrimitive.h

```diff
@@ -24,11 +24,13 @@
 #include <opencog/atoms/truthvalue/TruthValue.h>
 #include <opencog/guile/SchemeSmob.h>
 
-// Copied/pasted from gcc 4.9 utility. Remove as soon as C++14 is
-// enabled and gcc 4.9 is a minimum requirement.
+// Remove as soon as C++14 is enabled and gcc 4.9 or clang is a minimum requirement.
```

Comment:
Sooo ... we require gcc version 5 in `cogutils-git/cmake/OpenCogGccOptions.cmake`  so in fact, all of this can be removed, right? Or do you still need this for clang?

Reply:
ah, but we still ask for c++11 not c++ 14.  ... I'm going to change that now...

Reply:
I need this for clang right now because I'm telling clang to use C++11. Do you want me to try with C++14?

Reply:
I can confirm that I can build with c++14 with clang, with the same set of errors as c++11.

Reply:
That said... I noticed errors before I also recompiled cogutils with c++14. So other people could have mix-and-match issues with libraries using different c++ versions. (Sigh.) So I would tend to favour keeping c++11 compatibility while possible. You call, of course.

Reply:
Hmm. atomspace and opencog have almost no dependencies at all, just guile, python and boost, and there's no C++ in those first two. The ABI is supposed to guarantee that things compiled with c++11 can link & run correctly with things compiled with c++14.... Anyway, the compiler flags are in cogutils, anyway, so there is no way to compile the atomspace with different compiler flags, unless you hack them in.

My goal was to get rid of the integer-sequence code.